app/src/main/java/com/mobiledevelopmentfoodapp/service/RestaurantService.kt

```diff
@@ -10,11 +10,10 @@ import retrofit2.awaitResponse
 
 class RestaurantService {
     suspend fun fetchFoods(): List<Food>?{
-        return withContext(Dispatchers.IO){
+        return withContext(Dispatchers.IO) {
             val service = RetrofitClientInstance.retrofitInstance?.create(IFoodDAO::class.java)
-            val foods = async {service?.getAllFoods()}
-            var result =foods.await()?.awaitResponse()?.body()
-            return@withContext result
+            val foods = async { service?.getAllFoods() }
+            return@withContext foods.await()?.awaitResponse()?.body()
```

Comment:
Made variable return inline .

Reply:
There's a lot going on, on this line.  If something goes wrong and this line number is listed, it will be hard to identify where it went wrong.  I'd keep these as separate lines.